lib/safe_pusher/cli.rb

```diff
@@ -1,145 +1,73 @@
     end
 
     private
 
     attr_reader :arguments
 
     def execute_command(command)
-      if SHORTCUTS[command]
-        send(SHORTCUTS[command])
-      else
-        send(command)
-      end
-    end
-
-    def test
-      puts '##########################'.yellow
-      puts '## Testing new files... ##'.yellow
-      puts '##########################'.yellow
-      results = SafePusher::RspecRunner.new.call
-
-      exit results unless results == 0
-    end
-
-    def lint
-      puts '#######################'.yellow
-      puts '## Running linter... ##'.yellow
-      puts '#######################'.yellow
+      explain(command) if verbose
 
-      results = SafePusher::ProntoRunner.new.call
+      results = Kernel
+        .const_get("SafePusher::Client::#{services[command].capitalize}")
+        .new
+        .public_send(command)
 
       exit results unless results == 0
     end
 
-    def push
-      puts '##########################'.yellow
-      puts '## Pushing to Github... ##'.yellow
-      puts '##########################'.yellow
-
-      results = SafePusher::GithubRunner.new.push
+    def commands
+      @commands ||= arguments.map do |arg|
+        next unless arg =~ valid_commands_regexp
 
-      exit results unless results == 0
+        shortcut_to_command[arg] || arg
+      end
     end
 
-    def amend
-      puts '###################################'.yellow
-      puts '## Amending your last commit... ###'.yellow
-      puts '###################################'.yellow
-
-      results = SafePusher::GitRunner.new.amend
-
-      exit results unless results == 0
+    def version
+      puts SafePusher::VERSION
     end
 
-    def add
-      puts '######################'.yellow
-      puts '## Adding files... ###'.yellow
-      puts '######################'.yellow
-
-      results = SafePusher::GitRunner.new.add
-
-      exit results unless results == 0
+    def explain(command)
+      puts I18n.t("command.#{command}.verbose").yellow
     end
 
-    def commit
-      puts '################################'.yellow
-      puts '## Commiting last changes... ###'.yellow
-      puts '################################'.yellow
-
-      results = SafePusher::GitRunner.new.commit
-
-      exit results unless results == 0
+    def help
+      puts I18n.t('help')
     end
 
-    def open
-      puts '#########################################'.yellow
-      puts '## Opening a pull request on Github... ##'.yellow
-      puts '#########################################'.yellow
-
-      results = SafePusher::GithubRunner.new.open
-
-      exit results unless results == 0
+    def valid_commands_regexp
+      @valid_commands_regexp ||= /^(?!\s*$)(?:#{available_commands})$/
     end
 
-    def arguments_valid?
-      arguments.join(' ') =~ valid_commands_regexp
+    def available_commands
+      @available_commands ||= "#{shortcut_to_command.keys.join('|')}|"\
+        "#{shortcut_to_command.values.join('|')}"
```

Comment:
You can probably make only one call to `join` by doing something like:

`(shortcut_to_command.keys + shortcut_to_command.values).join('|')`